test/util/timer.test.cpp

```diff
@@ -131,6 +130,78 @@ TEST(Timer, TEST_REQUIRES_ACCURATE_TIMING(DestroyShouldStop)) {
     EXPECT_LE(totalTime, expectedTotalTime * 1.2);
 }
 
+
+TEST(Timer, TEST_REQUIRES_ACCURATE_TIMING(StoppedDuringExpiration)) {
+    // The idea is to have original timer cancellation and expiration roughly at the same time.
+    // In this case some timer backens (e.g. asio::high_resolution_timer) 
+    // may call the expiration callback with good status while the timer may not expect it.
+
+    RunLoop loop;
+
+    auto timer = std::make_unique<Timer>();
+    auto loopStopTimer = std::make_unique<Timer>();
+    auto expireTimeout = mbgl::Milliseconds(50);
+
+    auto timerCallback = [&] {
+        // we cannot expect much here as in some cases timer may be finished earlier
+        // than the loop stop timer (and thus the callback will be called)
+    };
+
+    auto loopStopTimerCallback = [&] {
+        timer->stop();
+        loop.stop();
+    };
+
+    auto first = mbgl::Clock::now();
+
+    loopStopTimer->start(expireTimeout, mbgl::Milliseconds(0),loopStopTimerCallback);
+    timer->start(expireTimeout, mbgl::Milliseconds(0), timerCallback);
+
+    loop.run();
+
+    auto totalTime = std::chrono::duration_cast<mbgl::Milliseconds>(mbgl::Clock::now() - first);
+
+    EXPECT_GE(totalTime, expireTimeout * 0.8);
```

Comment:
How did you determine these scaling factors? I'm a little afraid this might cause some false negatives on CI from time to time.

Reply:
All the timer tests in the file follow this pattern checking the range with these scaling factors in the end. I added these more for a consistency with all other tests assuming there are no major differences between the flavors of new and existing tests.